src/Npgsql/ConnectorPool.cs

```diff
@@ -404,7 +404,7 @@ await Task.WhenAny(tcs.Task, Task.Delay(timeLeft, delayCancellationToken.Token))
                         }
 
                         Debug.Assert(tcs.Task.IsCompleted);
-                        connector = tcs.Task.Result;
+                        connector = await tcs.Task;
```

Comment:
The task has already been awaited or waited above, so calling Result is more correct.